### ISSUE TITLE
Upgrading to puppeteer 10.2.0 breaks stylesheet rendering

### ISSUE THREAD

**Issue by jnimety** [Studiosity/grover#121]
Assets / Stylesheets aren't fetched after upgrading to Puppeteer 10.2.0. Puppeteer 10.1.0 works fine.

**Comment by jnimety** [comment]
As far as I can tell no assets are requested when using 10.2.0.

**Comment by abrom** [comment]
I've noticed some odd behaviour with 10.1.0 and 10.2.0 but haven't had a chance to look into it as yet.

**Comment by mkalygin** [comment]
@abrom I've done an investigation of the problem.

**Use case:** I want to produce PDF out of HTML string. Here is a minimal code for that scenario:

```ruby
require "grover"

Grover.configure do |config|
  config.options = {
    format: "A4",
    wait_until: "networkidle0",
    launch_args: [
      "--no-sandbox",
      "--disable-setuid-sandbox",
      "--disable-dev-shm-usage",
      "--single-process"
    ]
  }
end

html = <<-HTML
  <!DOCTYPE html>
  <html>
    <head>
      <base href="http://localhost:3000/">
      <link rel="stylesheet" type="text/css" href="test.css">
    </head>
    <body>
      <p>Test</p>
      <img src="test.jpg">
    </body>
  </html>
HTML

pdf = Grover.new(html).to_pdf
File.write("test.pdf", pdf)
File.write("test.html", html)
```

The output PDF doesn't have a stylesheet and an image loaded. It's broken.

I've found that the cause of it is the request interceptor's code:
https://github.com/Studiosity/grover/blob/main/lib/grover/js/processor.js#L145-L155

If I replace it with `page.setContent` it works as expected:
```javascript
requestOptions.waitUntil = waitUntil || 'networkidle0';
const displayUrl = options.displayUrl; delete options.displayUrl;
await page.goto(displayUrl || 'http://example.com', requestOptions);
await page.setContent(urlOrHtml, requestOptions);
```

**Comment by mkalygin** [comment]
@abrom I look into the release changes, and there are changes have been made to interceptor's functionality:
https://github.com/puppeteer/puppeteer/compare/v10.1.0...v10.2.0

I'm still investigating what can be a fix to this problem.

**Comment by abrom** [comment]
FYI this has been fixed in #122 and released in [v1.0.3](https://rubygems.org/gems/grover/versions/1.0.3)

**Comment by jnimety** [comment]
Thank you!
